File: _code-samples/checks/py/create_check.py
from datetime import datetime, timedelta

from xrpl.clients import JsonRpcClient
from xrpl.models import CheckCreate, IssuedCurrencyAmount
from xrpl.transaction import submit_and_wait
from xrpl.utils import datetime_to_ripple_time, xrp_to_drops
from xrpl.wallet import generate_faucet_wallet

client = JsonRpcClient("https://s.altnet.rippletest.net:51234") # Connect to the testnetwork

"""Create a token check"""

check_receiver_addr = "rPT1Sjq2YGrBMTttX4GZHjKu9dyfzbpAYe" # Example: send back to Testnet Faucet

token_name = "USD" 

amount_to_deliver = 10.00

token_issuer = "r9CEVt4Cmcjt68ME6GKyhf2DyEGo2rG8AW"

# Set check to expire after 5 days
expiry_date = datetime_to_ripple_time(datetime.now() + timedelta(days=5))

# Generate wallet
sender_wallet = generate_faucet_wallet(client=client)

# Build check create transaction
check_txn = CheckCreate(account=sender_wallet.address, destination=check_receiver_addr,
                        send_max=IssuedCurrencyAmount(
                            currency=token_name,
                            issuer=token_issuer,
                            value=amount_to_deliver),
                        expiration=expiry_date)

# Autofill, sign, then submit transaction and wait for result
stxn_response = submit_and_wait(check_txn, client, sender_wallet)

# Parse response for result
stxn_result = stxn_response.result

# Print result and transaction hash
print(stxn_result["meta"]["TransactionResult"])
print(stxn_result["hash"])


############### CREATE XRP CHECK ################################
"""Create xrp check"""


check_receiver_addr = "rPT1Sjq2YGrBMTttX4GZHjKu9dyfzbpAYe" # Example: send back to Testnet Faucet

amount_to_deliver = 10.00

# Set check to expire after 5 days
expiry_date = datetime_to_ripple_time(datetime.now() + timedelta(days=5))


# Generate wallet
sender_wallet = generate_faucet_wallet(client=client)

# Build check create transaction
check_txn = CheckCreate(account=sender_wallet.address,
                        destination=check_receiver_addr,
                        send_max=xrp_to_drops(amount_to_deliver),
                        expiration=expiry_date)

# Autofill, sign, then submit transaction and wait for result
stxn_response = submit_and_wait(check_txn, client, sender_wallet)

# Parse response for result
stxn_result = stxn_response.result

# Print result and transaction hash
print(stxn_result["meta"]["TransactionResult"])
print(stxn_result["hash"])

File: _code-samples/checks/js/create-check.js
import xrpl from 'xrpl'

// Connect to the Testnet -------------------------------------------------

console.log('Connecting to Testnet...')
const client = new xrpl.Client('wss://s.altnet.rippletest.net:51233')
await client.connect()
console.log('Connected.')

// Get a new wallet -------------------------------------------------------

console.log('Generating new wallet...')
const wallet = (await client.fundWallet()).wallet
console.log('  Address:', wallet.address)
console.log('  Seed:', wallet.seed)

// Prepare the transaction ------------------------------------------------

const checkCreate = {
  TransactionType: 'CheckCreate',
  Account: wallet.address,
  Destination: 'rGPnRH1EBpHeTF2QG8DCAgM7z5pb75LAis',
  SendMax: xrpl.xrpToDrops(120), // Can be more than you have
  InvoiceID: '46060241FABCF692D4D934BA2A6C4427CD4279083E38C77CBE642243E43BE291'
}

// Validate the transaction before submitting -----------------------------

xrpl.validate(checkCreate)
console.log(JSON.stringify(checkCreate, null, 2))

// Submit the transaction -------------------------------------------------

console.log('Submitting transaction...')
const tx = await client.submitAndWait(
  checkCreate,
  { autofill: true,
    wallet }
)

// Confirm transaction result and get check ID ------------------------------------

const resultCode = tx.result.meta.TransactionResult
if (resultCode !== 'tesSUCCESS') {
  console.error('Unable to create check:', resultCode)
  await client.disconnect()
  process.exit(1)
}

const node = tx.result.meta.AffectedNodes.find(node =>
   node.CreatedNode?.LedgerEntryType === 'Check'
  ).CreatedNode

console.log('Check created successfully.')
console.log(`Check details:\n`, JSON.stringify(node.NewFields, null, 2))
console.log(`Check ID: ${node.LedgerIndex}`)

// Disconnect -------------------------------------------------------------

await client.disconnect()
